src/main/java/seedu/address/model/person/PersonContainsKeywordsPredicate.java

```diff
@@ -0,0 +1,34 @@
+package seedu.address.model.person;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import seedu.address.model.tag.Tag;
+
+
+/**
+ * Tests that a {@code ReadOnlyPerson}'s {@code Tag} matches any of the keywords given.
+ */
+public class PersonContainsKeywordsPredicate implements Predicate<ReadOnlyPerson> {
+    private final List<Tag> keywords;
+
+    public PersonContainsKeywordsPredicate(List<Tag> keywords) {
+        this.keywords = keywords;
+    }
+
+    @Override
+    public boolean test(ReadOnlyPerson person) {
+        return keywords.stream()
+                .anyMatch(keyword -> person.getTags().contains(keyword));
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof PersonContainsKeywordsPredicate // instanceof handles nulls
+                && this.keywords.equals(((PersonContainsKeywordsPredicate) other).keywords)); // state check
```

Comment:
here should include additional indentation as `this.keywords.equals(((PersonContainsKeywordsPredicate) other).keywords)` in this boolean statement is not on the same level as `this`